### PR TITLE
python: add terminal output parser of tables

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/RRZE-HPC/likwid/issues/199

@phibel I checked this with `likwid-mpirun` and `MEM_DP` group with markers, but I know for sure in it's current form it won't work for `likwid-perfct` as it has a different output format for regions and groups. Hopefully this script can be improved/bended to cover that case, as well as `likwid-powermeter`. Feel free to create PRs to this branch.

### CODE REVIEW

**Comment by davydden** [comment]
@TomTheBear I looked at the `CVS` output, although it's easier to parse, it's still not user friendly in that one needs to manually separate data for different tables (specific regions, sum or events, etc). 

Looking at the output, I have a feeling that it is **not** easier to parse because the output completely lacks any separators of actual tables. Namely it is just a single huge _table_ (actually not):

```
Region,blabla,
Group,1,
Event,Counter,e0107:0:0
Runtime (RDTSC) [s],TSC,1.205799
Region calls,CTR,10
INSTR_RETIRED_ANY,FIXC0,5067253000
...
CAS_COUNT_RD,MBOX7C0,0
CAS_COUNT_WR,MBOX7C1,0
Metric,e0107:0:0,
Runtime (RDTSC) [s],1.2058,
...
Memory data volume [GBytes],9.5893,
Operational intensity,1.0240,
Region,BLABLABLA,
Group,1,
Event,Counter,e0107:0:0
...
```

One could of course add some keywords, i.e. we know that `Region` and `Group` are NOT tables.
But then I would need to add all possible keywords for other table headers, i.e. `Event` and `Metric`. 
So I am not yet sure if it's easier to parse CVS or terminal output. Terminal output at least has separators for start/end of the actual table:
```
+--------------------------------------+---------+-------------+
|                 Event                | Counter |  e1109:0:0  |
+--------------------------------------+---------+-------------+
BLABLA
+--------------------------------------+---------+-------------+
```
so it's easier to catch the header of the table and the actual content without extra assumptions. 

Bottom line is, I think python parser is still userful as it allows one to work with parsed results via
```
result['vmult']['Metric_Sum']['MFLOP/s']
```

For my needs, I would probably stick with parsing the beautified terminal output.

**Comment by TomTheBear** [comment]
Hi,

I understand the need of such a parser but I don't see your arguments against the CSV output. It actually contains keywords already (STRUCT and TABLE):
```
STRUCT,Info,3,,,
CPU name:,Intel(R) Core(TM) i7-4770 CPU @ 3.40GHz,,,,
CPU type:,Intel Core Haswell processor,,,,
CPU clock:,3.392181359 GHz,,,,
TABLE,Region copy,Group 1 Raw,L3,5,
Region Info,Core 0,Core 1,,,
RDTSC Runtime [s],0.490816,0.461983,,,
call count,10,10,,,
Event,Counter,Core 0,Core 1,,
INSTR_RETIRED_ANY,FIXC0,602320300,600771400,,
CPU_CLK_UNHALTED_CORE,FIXC1,1503458000,1498534000,,
CPU_CLK_UNHALTED_REF,FIXC2,1503456000,1498554000,,
L2_LINES_IN_ALL,PMC0,49862770,50043380,,
L2_TRANS_L2_WB,PMC1,15348130,15296480,,
TABLE,Region copy,Group 1 Raw STAT,L3,5,
[...]
```

**Comment by davydden** [comment]
@TomTheBear 

> It actually contains keywords already (STRUCT and TABLE):

which version of LIKWID is that? I don't have those keywords in the output from 4.2.1.

**Comment by TomTheBear** [comment]
A more recent one (4.3.3) ;) The markers in CSV were added in May 2015 (https://github.com/RRZE-HPC/likwid/commit/ea60dda3f60a6724601232e571510a6ffc1a1ec2)

**Comment by davydden** [comment]
> A more recent one (4.3.3) ;) The markers in CSV were added in May 2015 ([ea60dda](https://github.com/RRZE-HPC/likwid/commit/ea60dda3f60a6724601232e571510a6ffc1a1ec2))

👍 

ah, ok, let me try with the most recent version and see what native python parsers do with such output...

**Comment by davydden** [comment]
@TomTheBear 4.3.3. is not there on Emmy, whereas 4.3.2 does NOT contain keywords on CVS output. At least not from 
```
likwid-mpirun -np 1 -nperdomain S:10 -g MEM_DP -m -O
```

I will until 4.3.3. is available to see how CVS works and if it's easier to parse. Will keep the PR open for now...
